cli/command_restore.go

```diff
@@ -0,0 +1,81 @@
+package cli
+
+import (
+	"context"
+	"io"
+	"os"
+	"path"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/object"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+var (
+	restoreSnapshotID = restoreCommands.Arg("snapshot-id", "Snapshot ID from which to restore.").Required().String()
+	restoreTargetPath = restoreCommands.Arg("target-path", "Path to which the snapshot must be restored.").Required().String()
+)
+
+func runRestoreCommand(ctx context.Context, rep *repo.Repository) error {
+	oid, err := parseObjectID(ctx, rep, *restoreSnapshotID)
+	if err != nil {
+		return err
+	}
+	targetPath := *restoreTargetPath
+
+	return restoreRecursively(ctx, rep, targetPath, oid)
+}
+
+func init() {
+	restoreCommands.Action(repositoryAction(runRestoreCommand))
+}
+
+func restoreRecursively(ctx context.Context, rep *repo.Repository, targetPath string, oid object.ID) error {
+	d := snapshotfs.DirectoryEntry(rep, oid, nil)
+
+	entries, err := d.Readdir(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range entries {
+		restorePath := path.Join(targetPath, e.Name())
+		objectID := e.(object.HasObjectID).ObjectID()
+		// Restore directories recursively
+		if e.Mode().IsDir() {
+			// Create the directory
+			if direrr := os.MkdirAll(restorePath, 0777); direrr != nil {
+				return direrr
+			}
+			if recerr := restoreRecursively(ctx, rep, restorePath+"/", objectID); recerr != nil {
+				return recerr
+			}
+			// Set permissions as stored in the backup
```

Comment:
In this context, the term "snapshot" should be used instead of "backup", in order to be consistent with the terminology used in Kopia
```suggestion
			// Set permissions as stored in the snapshot
```
Same below